File: src/main/java/com/kkllffaa/meteor_litematica_printer/Modules/Tools/MidiParser.java
package com.kkllffaa.meteor_litematica_printer.Modules.Tools;

import com.kkllffaa.meteor_litematica_printer.Addon;

import meteordevelopment.meteorclient.settings.*;
import meteordevelopment.meteorclient.systems.modules.Module;



public class MidiParser extends Module {

    public MidiParser() {
        super(Addon.TOOLSCATEGORY, "midi-parser", "Parses MIDI files.");
    }
    private final SettingGroup sgGeneral = settings.getDefaultGroup();

    private final Setting<String> midiFilePath = sgGeneral.add(new StringSetting.Builder()
        .name("midi-file-path")
        .description("Path to the MIDI file to parse.")
        .defaultValue("")
        .build()
    );

    private final Setting<Boolean> parseMidi = sgGeneral.add(new BoolSetting.Builder()
        .name("parse-midi")
        .description("Click to parse the MIDI file.")
        .defaultValue(false)
        .onChanged(this::onParseMidiChanged)
        .build()
    );

    private void onParseMidiChanged(Boolean value) {
        if (value) {
            parseMidiFile();
            parseMidi.set(false); // reset
        }
    }

    private void parseMidiFile() {
        String path = midiFilePath.get();
        if (path.isEmpty()) {
            error("MIDI file path is empty.");
            return;
        }
        info("Parsing MIDI file: " + path);
       
    }

}

File: src/main/kotlin/com/kkllffaa/meteor_litematica_printer/Modules/Tools/MidiParser.kt
package com.kkllffaa.meteor_litematica_printer.Modules.Tools

import com.kkllffaa.meteor_litematica_printer.Addon
import meteordevelopment.meteorclient.settings.BoolSetting
import meteordevelopment.meteorclient.settings.Setting
import meteordevelopment.meteorclient.settings.StringSetting
import meteordevelopment.meteorclient.systems.modules.Module

class MidiParser : Module(Addon.TOOLS, "midi-parser", "Parses MIDI files.") {
    private val sgGeneral = settings.defaultGroup

    private val midiFilePath: Setting<String> = sgGeneral.add(
        StringSetting.Builder()
            .name("midi-file-path")
            .description("Path to the MIDI file to parse.")
            .defaultValue("")
            .build()
    )

    private val parseMidi: Setting<Boolean> = sgGeneral.add(
        BoolSetting.Builder()
            .name("parse-midi")
            .description("Click to parse the MIDI file.")
            .defaultValue(false)
            .onChanged { value: Boolean -> this.onParseMidiChanged(value) }
            .build()
    )

    private fun onParseMidiChanged(value: Boolean) {
        if (value) {
            parseMidiFile()
            parseMidi.set(false) // reset
        }
    }

    private fun parseMidiFile() {
        val path = midiFilePath.get()
        if (path.isEmpty()) {
            error("MIDI file path is empty.")
            return
        }
        info("Parsing MIDI file: $path")
    }
}
